File: arena-pytest/tests/playbooks/calibration_outage.py
from arena_pytest import HttpPlaybookBuilder, ManagedHttpPlaybook, server_error

from readings_arena_config import (
    CALIBRATION_VALIDATE_PATH,
    PLAYBOOK_CALIBRATION_OUTAGE_MANAGED,
)


class CalibrationOutagePlaybook(ManagedHttpPlaybook):
    def __init__(self, dependency_identifier: str):
        mappings = (
            HttpPlaybookBuilder(dependency_identifier)
            .post(CALIBRATION_VALIDATE_PATH)
            .will_return(server_error())
            .expect_called_at_least(1)
            .into_playbook()
            .mappings_for_ffi()
        )
        super().__init__(
            identifier=PLAYBOOK_CALIBRATION_OUTAGE_MANAGED,
            dependency_identifier=dependency_identifier,
            mappings=mappings,
        )

File: examples/tests/example-readings-axum-component-test/playbooks/calibration_outage.rs
use arena_http::{server_error, ManagedHttpPlaybook};

use super::ids::{calibration_outage_managed_id, calibration_validate_path};

pub fn calibration_outage_playbook(calibration_id: impl Into<String>) -> ManagedHttpPlaybook {
    ManagedHttpPlaybook::new(calibration_outage_managed_id(), calibration_id, |pb| {
        pb.post(calibration_validate_path())
            .will_return(server_error())
            .expect_called_at_least(1)
            .into_playbook()
    })
}
